rabbitmq/broker.go

```diff
@@ -175,70 +176,68 @@ func (br *Broker) Observe(queues []string) {
 // Receive fetches a Celery task message from a tail of one of the queues in RabbitMQ.
 // After a timeout it returns nil, nil.
 func (br *Broker) Receive() ([]byte, error) {
-
-	const retryIntervalMs = 100
-
-	try_receive := func() (msg amqp.Delivery, ok bool, err error) {
-		queue := br.queues[0]
-		// Put the Celery queue name to the end of the slice for fair processing.
-		broker.Move2back(br.queues, queue)
-		my_msg, my_ok, my_err := br.channel.Get(queue, true)
-		if my_err != nil {
-			log.Printf("Failed to g a message: %s", my_err)
+	queue := br.queues[0]
+	// Put the Celery queue name to the end of the slice for fair processing.
+	broker.Move2back(br.queues, queue)
+
+	delivery, ok := br.delivery[queue]
+	if !ok {
+		delivery, err := br.channel.Consume(
+			queue, // queue
+			"",    // consumer
+			true,  // autoAck
+			false, // exclusive
+			false, // noLocal (ignored)
+			false, // noWait
+			nil,   // args
+		)
+		if err != nil {
+			err_str := fmt.Errorf("%w", err)
+			log.Printf("channel.Consume(): %s", err_str)
+			return nil, nil
```

Comment:
Why not return an error here?

Reply:
I ended up moving this code to the Observe() function, where it's safe to call log.Panicf() if any error occurs.